capstone-backend/src/main/java/com/google/sps/data/Assignment.java

```diff
@@ -0,0 +1,69 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Assignment extends BaseEntity {
+  private String text;
+  private final String clubID;
+  private List<Comment> comments;
+  private final String whenCreated;
+  private String whenDue;
+
+  public Assignment(String text, String clubID, List<Comment> comments, String whenCreated,
+      String whenDue) {
+    this.text = text;
+    this.clubID = clubID;
+    this.comments = comments;
+    this.whenCreated = whenCreated;
+    this.whenDue = whenDue;
+  }
+
+  public Assignment(String text, String clubID, String whenCreated, String whenDue) {
+    this(text, clubID, new ArrayList<Comment>(), whenCreated, whenDue);
+  }
+
+  public Assignment() {
+    this("", "", "", "");
+  }
+
+  public String getText() {
+    return text;
+  }
+
+  public void setText(String text) {
+    this.text = text;
+  }
+
+  public String getClubID() {
+    return clubID;
+  }
+
+  public List<Comment> getComments() {
+    return comments;
+  }
+
+  public void setComments(ArrayList<Comment> comments) {
```

Comment:
I still don't know why we need setters for all objects? Do you know when in the Firestore documentation its say they are required?

Reply:
If you don't include getters and setters for objects, Firestore will throw errors that object fields "cannot be serialized." While they are not explicitly called in our code, Firestore uses them to create and update object fields and we need to include them.